frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDetailsVersionContext.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import usePipelineVersionById from '~/concepts/pipelines/apiHooks/usePipelineVersionById';
+import { getPipelineVersionResourceRef } from '~/concepts/pipelines/content/tables/utils';
+import {
+  PipelineRunJobKF,
+  PipelineRunKF,
+  PipelineVersionKF,
+  ResourceReferenceKF,
+} from '~/concepts/pipelines/kfTypes';
+
+type PipelineRunDetailsVersionContextState = {
+  version: PipelineVersionKF | null;
+  loaded: boolean;
+  error?: Error;
+  versionRef?: ResourceReferenceKF;
+};
+
+export const PipelineRunDetailsVersionContext =
+  React.createContext<PipelineRunDetailsVersionContextState>({
+    version: null,
+    loaded: false,
+    error: undefined,
+    versionRef: undefined,
+  });
+
+type PipelineRunDetailsVersionContextProps = {
+  children: React.ReactNode;
+  resource?: PipelineRunJobKF | PipelineRunKF | null;
```

Comment:
Yes, I need to do some refactor on the pipeline details page because the run is from `const run = runResource?.run;`, which could be `undefined` and the early exit should be checked. I will change that.